tests/json_schema.go

```diff
@@ -48,98 +59,48 @@ func TestPayloadAttributesInSchema(t *testing.T, name string, undocumentedAttrs
 	}
 }
 
-func TestJsonSchemaKeywordLimitation(t *testing.T, fieldPaths []string, schema string, exceptions *set.Set) {
-	fieldNames, err := fetchFlattenedFieldNames(fieldPaths, addKeywordFields)
-	assert.NoError(t, err)
-
-	schemaStruct, _ := schemaStruct(strings.NewReader(schema))
-	schemaNames := set.New()
-	flattenSchemaNames(schemaStruct, "", addLengthRestrictedPropNames, schemaNames)
-
-	mapping := []Mapping{
-		{"errors.context", "context"},
-		{"transactions.context", "context"},
-		{"errors", "error"},
-		{"transactions.traces", "trace"},
-		{"transactions", "transaction"},
-		{"app", "context.app"},
-		{"system", "context.system"},
-	}
-
-	mappedSchemaNames := set.New()
-	for _, k := range schemaNames.List() {
-		name := k.(string)
-		for _, m := range mapping {
-			if strings.HasPrefix(name, m.from) {
-				k = strings.Replace(name, m.from, m.to, 1)
-				break
-			}
-		}
-		mappedSchemaNames.Add(k)
-	}
-	found := set.Union(mappedSchemaNames, exceptions).(*set.Set)
-	diff := set.SymmetricDifference(fieldNames, found).(*set.Set)
-
-	errMsg := fmt.Sprintf("Missing json schema length limit 1024 for: %v", diff)
-	assert.Equal(t, 0, diff.Size(), errMsg)
-}
-
-func schemaStruct(reader io.Reader) (*Schema, error) {
-	decoder := json.NewDecoder(reader)
-	var schema Schema
-	err := decoder.Decode(&schema)
-	return &schema, err
-}
-
-func flattenSchemaNames(s *Schema, prefix string, addFn addProperty, flattened *set.Set) {
-	if len(s.Properties) > 0 {
-		for k, v := range s.Properties {
-			flattenedKey := StrConcat(prefix, k, ".")
-			if addFn(v) {
-				flattened.Add(flattenedKey)
-			}
-			flattenSchemaNames(v, flattenedKey, addFn, flattened)
-		}
-	} else if s.Items != nil {
-		flattenSchemaNames(s.Items, prefix, addFn, flattened)
+// loads jsonSchema file (.json) given a current directory, and returns its contents and directory
+// current directories need to be propagate across calls to loadSchema because $ref paths are relative
+func loadSchema(spec string, currentDir string) (map[string]interface{}, string) {
+	if currentDir == "" {
+		_, current, _, _ := runtime.Caller(0)
+		currentDir = filepath.Join(filepath.Dir(current), "../docs/spec/")
 	}
+	specFile := filepath.Join(currentDir, spec)
+	schemaData, _ := ioutil.ReadFile(specFile)
+	var schema map[string]interface{}
+	json.Unmarshal(schemaData, &schema)
+	return schema, filepath.Dir(specFile)
 }
 
-func flattenJsonKeys(data interface{}, prefix string, flattened *set.Set) {
+// walks a json collecting nested keys separated by dots, ie. {a:{b:c:{d:1}} ==> a.b.c.d
+// accumulates the result in the first argument, which is mutated recursively
+// it has some rudimentary knowledge of jsonSchema.
+// jsonSchema keys are traversed, but not accumulated in the result
+func flattenJsonKeys(flattened *set.Set, data interface{}, currentSpecDir string, keyPrefix string) {
```

Comment:
Is it correctly understood that this now is intended to be used both for payloads and for JSON Schema files?

Reply:
yes